Makefile

```diff
@@ -16,6 +16,9 @@ bench: $(subdirs)
 check:  $(subdirs)
 	$(MAKE) -C test $@
 
+valgrind-check:  $(subdirs)
+	$(MAKE) -C test check TEST_WRAPPER="valgrind --leak-check=full --suppressions=valgrind.supp"
```

Comment:
I think `--error-exitcode=1` could also be helpful here, to guarantee valgrind exits with a non-zero error code when it finds something odd.